scheduler/replica_scheduler.go

```diff
@@ -110,6 +110,17 @@ func (rcs *ReplicaScheduler) filterNodeDisksForReplica(node *longhorn.Node, repl
 			logrus.Errorf("Fail to get settings when scheduling replica: %v", err)
 			return preferredDisk
 		}
+		scheduledReplica := status.ScheduledReplica
+		// check other replicas for the same volume has been accounted on current node
+		var storageScheduled int64
+		for rName, r := range replicas {
+			if _, ok := scheduledReplica[rName]; !ok && r.Spec.NodeID != "" && r.Spec.NodeID == node.Name {
+				storageScheduled += r.Spec.VolumeSize
```

Comment:
If the replica schedule before replica map update, it will cause the schedule issue. So I add a map check, if the replica in the same volume has been scheduled but not save to node status, I will add this part of storage space to scheduler.

Reply:
How would it impact the system if we just return here and retry if the scheduled storage wasn't updated? I think the logic is sound but separating the storage accounting logic in two places make things more complex.